validation/validate.py

```diff
@@ -0,0 +1,215 @@
+import yaml
+import pytest
+import re
+from typing import Any
+
+
+def read_yaml_file(file_name: str) -> str:
+    '''
+    Reads a YAML file and returns it as a dictionary.
+    '''
+    with open(file_name, 'r') as f:
+        try:
+            d = yaml.safe_load(f)
+        except yaml.YAMLError as e:
+            print(e)
+    return d
+
+
+def type_matches(value: Any, expected_type: str) -> bool:
+    '''
+    Checks whether value has the type expected_type.
+    If yes, returns True, otherwise False.
+
+    Allowed types T are: 'str', 'int', 'float', 'complex', 'bool',
+                         as well as 'List[T]'.
+
+    For any other types the function raises ValueError.
+    '''
+    allowed_basic_types = ['str', 'int', 'float', 'complex', 'bool']
+    allowed_list_types = ['List[{}]'.format(t) for t in allowed_basic_types]
+    allowed_types = allowed_basic_types + allowed_list_types
+
+    # first verify whether expected_type is allowed
+    if expected_type not in allowed_types:
+        raise ValueError('could not recognize expected_type: {}'.format(expected_type))
+
+    expected_type_is_list = re.match(r"^List\[\w+\]$", expected_type) is not None
+
+    if expected_type_is_list:
+        # make sure that value is actually a list
+        if not type(value).__name__ == 'list':
```

Comment:
Can `isinstance` be used instead?

Reply:
Yes, I think `isinstance` can be used - I will adjust.

Reply:
I will first move/split the files, enable testing, and then I will replace the `isinstance`. So it's not forgotten but I don't want to change this blindly - yesterday I was testing interactively, now I first want to set up automated tests for the many checks before touching the code.

Reply:
Aha - now I remember why I did not use isinstance: I wanted to compare the string:
```
# wrong but would be convenient for me
isinstance(3, 'int')

# correct but inconvenient for me
isinstance(3, int)
```
I am sure this can be somehow solved but how about I place a comment into the code and I or somebody else can be later creative about it?

Reply:
Go for the comment! Probably we need to put a `lambda` there that returns the type given the string.